Mastodon/Scene/Settings/Privacy and Safety/PrivacySafetyViewModel.swift

```diff
@@ -0,0 +1,217 @@
+        case .custom:
+            break
+        }
+    }
+    
+    func evaluatePreset() {
+        guard !doNotEvaluate else { return }
+        if self == Self.openPublic {
+            preset = .openPublic
+        } else if self == Self.privateRestricted {
+            preset = .privateRestricted
+        } else {
+            preset = .custom
+        }
+    }
+
+    func loadSettings() {
+        Task { @MainActor in
+            guard let appContext, let authContext else {
+                return dismiss()
+            }
+            
+            let domain = authContext.mastodonAuthenticationBox.domain
+            let userAuthorization = authContext.mastodonAuthenticationBox.userAuthorization
+            
+            let account = try await appContext.apiService.accountVerifyCredentials(
+                domain: domain,
+                authorization: userAuthorization
+            ).singleOutput().value
+            
+            if let privacy = account.source?.privacy {
+                visibility = .from(privacy)
+            }
+            
+            manuallyApproveFollowRequests = account.locked == true
+            showFollowersAndFollowing = account.hideCollections == false
+            suggestMyAccountToOthers = account.discoverable == true
+            appearInSearches = account.indexable == true
+
+            isUserInteractionEnabled = true
+        }
+    }
+    
+    func saveSettings() {
+        Task {
+            guard let appContext, let authContext else {
+                return
+            }
+    
+            let domain = authContext.mastodonAuthenticationBox.domain
+            let userAuthorization = authContext.mastodonAuthenticationBox.userAuthorization
+            
+            let _ = try await appContext.apiService.accountUpdateCredentials(
+                domain: domain,
+                query: .init(
+                    discoverable: suggestMyAccountToOthers,
+                    locked: manuallyApproveFollowRequests,
+                    source: .withPrivacy(visibility.toPrivacy()),
+                    indexable: appearInSearches,
+                    hideCollections: !showFollowersAndFollowing
+                ),
+                authorization: userAuthorization
+            ).value
+        }
+    }
+    
+    func dismiss() {
+        onDismiss.send(())
+    }
+}
+
+// Preset Rules Definition
+extension PrivacySafetyViewModel {
+    static let openPublic: PrivacySafetyViewModel = {
+        let vm = PrivacySafetyViewModel(appContext: nil, authContext: nil, coordinator: nil)
+        vm.visibility = .public
+        vm.manuallyApproveFollowRequests = false
+        vm.showFollowersAndFollowing = true
+        vm.suggestMyAccountToOthers = true
+        vm.appearInSearches = true
+        return vm
+    }()
+    
+    static let privateRestricted: PrivacySafetyViewModel = {
+        let vm = PrivacySafetyViewModel(appContext: nil, authContext: nil, coordinator: nil)
```

Comment:
I needed a moment to understand, why the parameters are nil, but from my understanding, `openPublic` and `privateRestricted` both serve as templates, right? Their `appContext` etc. is totally irrelevant, as their other values get applied to a `PrivacySafetyViewModel`, which already has a `appContext, ...`, right?

Reply:
Yes these defintions are only used for comparing agains the current settings and applying presets, so the context here is not consumed. Now that I'm writing this, I think the cleaner solution would be to use protocols for this, I'll fix this.

Reply:
And it's done, thanks for the Feedback!